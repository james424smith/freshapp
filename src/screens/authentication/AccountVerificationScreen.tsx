import React, { useEffect, useState } from "react";
import TermsConditionsPrivacyPolicyLinks from "../../components/termsAndPrivacyPolicy/TermsAndPrivacyPolicyLink";
import { View, TouchableOpacity } from "react-native";
import Text from "../../components/StyledText";
import { Button, Input } from "react-native-elements";
import InputElement from "../../components/Input";
import styles from "../../styles/AuthStyles";
import { RESET_PASSWORD_ROUTE } from "../../constants/routes";
import Colors from "../../constants/styles/colors";
import { RouteProp, useNavigation, useTheme } from "@react-navigation/native";
import EyeOff from "../../../assets/icons/eye-off.svg";
import EyeOn from "../../../assets/icons/eye-on.svg";
import AsyncStorage from "@react-native-community/async-storage";
import { AuthStackParamList } from "../../navigation/AppNavigator";
import ExitButton from "../../components/authentication/ExitButton";
import AuthMainView from "../../components/authentication/AuthMainView";

interface Props {
  route: RouteProp<AuthStackParamList, "AccountVerification">;
}
const AccountVerificationScreen = (props: Props) => {
  const codeInput = React.useRef<Input | null>(null);
  const theme = useTheme();
  const [employeeId, setEmployeeId] = useState<string>("");
  const [code, setCode] = useState<string>("");
  const [showLoading, setShowLoading] = useState<boolean>(false);
  const navigation = useNavigation();
  const [shouldShowPassword, setShouldShowPassword] = useState<boolean>(false);
  const { route } = props;
  const errorMessage = route.params?.verificationError;

  useEffect(() => {
    getEmployeeIdFromStorage();
  }, []);

  const getEmployeeIdFromStorage = async () => {
    const storageEmployeeId = (await AsyncStorage.getItem("employeeID")) || "";
    setEmployeeId(storageEmployeeId);
  };

  const renderErrorMessage = () => {
    if (errorMessage) {
      return (
        <View style={styles.errorMessageContainer}>
          <Text style={styles.errorMessageText}>{errorMessage}</Text>
        </View>
      );
    }
    return null;
  };

  const mainContent = () => {
    return (
      <View style={styles.resetPasswordLayout}>
        <ExitButton path={"SIGN_IN"} />
        <View />
        <View style={styles.resetForgotPasswordTitleContainer}>
          <Text style={styles.resetPasswordText}>Account verification</Text>
        </View>
        <View style={styles.forgotPasswordCredentialsContainer}>
          <View>
            <Text
              style={{
                ...styles.accountVerificationEmployeeText,
                color: theme.colors.accountVerificationID,
              }}
            >
              Employee ID
            </Text>
          </View>
          <InputElement
            testID={"employee-id-input"}
            containerStyle={{
              ...styles.inputBoxStyle,
              backgroundColor: theme.colors.colorInputText,
            }}
            onChangeText={(newEmployeeId: string) =>
              setEmployeeId(newEmployeeId)
            }
            onSubmitEditing={() => {
              codeInput.current && codeInput.current.focus();
            }}
            value={employeeId}
            inputStyle={styles.inputValueStyle}
            keyboardAppearance="light"
            placeholder="Employee ID"
            autoFocus={false}
            autoCapitalize="none"
            autoCorrect={false}
            keyboardType="numeric"
            returnKeyType="default"
            blurOnSubmit={false}
            placeholderTextColor={Colors.FAQInactiveColor}
            inputContainerStyle={styles.removeUnderline}
            shouldUseFocusedStyles={true}
            disabled={true}
          />
          <InputElement
            testID={"code-input"}
            containerStyle={{
              ...styles.inputBoxStyle,
              backgroundColor: theme.colors.colorInputText,
            }}
            secureTextEntry={!shouldShowPassword}
            onChangeText={(newCode: string) => setCode(newCode)}
            value={code}
            inputStyle={styles.inputValueStyle}
            keyboardAppearance="light"
            placeholderTextColor={Colors.FAQInactiveColor}
            placeholder="Email Verification code "
            autoCapitalize="none"
            inputContainerStyle={[
              styles.removeUnderline,
              styles.centerInputText,
            ]}
            autoCorrect={false}
            keyboardType="numeric"
            returnKeyType="next"
            ref={codeInput}
            shouldUseFocusedStyles={true}
            blurOnSubmit={false}
            rightIcon={
              <TouchableOpacity
                testID={"show-password-icon"}
                onPress={() => setShouldShowPassword(!shouldShowPassword)}
              >
                {shouldShowPassword ? (
                  <EyeOn width={20} height={20} fill={"#000"} />
                ) : (
                  <EyeOff width={20} height={20} fill={"#fff"} />
                )}
              </TouchableOpacity>
            }
          />
          {renderErrorMessage()}
          <View style={styles.cancelLinkContainer}>
            <Button
              testID={"submit-button"}
              title="Next"
              activeOpacity={1}
              onPress={() => {
                setShowLoading(false);
                navigation.navigate(RESET_PASSWORD_ROUTE, {
                  employeeId,
                  verificationCode: code,
                });
              }}
              loading={showLoading}
              buttonStyle={{
                ...styles.signUpButton,
                backgroundColor: theme.colors.signInButtonColor,
              }}
              loadingProps={{
                size: "small",
                color: Colors.white,
              }}
              disabled={!employeeId || !code}
              disabledStyle={{
                backgroundColor: theme.colors.DisableSignInButtonColor,
              }}
              titleStyle={styles.textOnButton}
              disabledTitleStyle={{
                color: theme.colors.whiteAndBlack,
              }}
            />
          </View>
        </View>
        <View style={styles.forgotPasswordTermsConditionsContainer}>
          <View style={styles.termsConditionsView}>
            <TermsConditionsPrivacyPolicyLinks navigate={navigation.navigate} />
          </View>
        </View>
      </View>
    );
  };

  return <AuthMainView>{mainContent()}</AuthMainView>;
};
export default AccountVerificationScreen;
