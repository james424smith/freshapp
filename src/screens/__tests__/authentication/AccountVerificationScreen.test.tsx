import React from "react";
import AccountVerificationScreen from "../../authentication/AccountVerificationScreen";
import {
  render,
  cleanup,
  fireEvent,
  waitFor,
} from "@testing-library/react-native";

describe("AccountVerificationScreen snapshot test", () => {
  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
  });

  it("should render correct ui for AccountVerificationScreen when verificationError is undefined", async () => {
    const route = {
      key: "AccountVerification-wPFrEJG5kjiOrE_Qk2kVs",
      name: "AccountVerification" as "AccountVerification",
      params: { verificationError: undefined },
    };

    const { toJSON, getByTestId } = render(
      <AccountVerificationScreen route={route} />
    );

    fireEvent.press(getByTestId("show-password-icon"));

    fireEvent.changeText(getByTestId("employee-id-input"), "some");
    fireEvent(getByTestId("employee-id-input"), "onSubmitEditing", {
      nativeEvent: { text: "some" },
    });
    fireEvent.changeText(getByTestId("code-input"), "code");
    fireEvent.press(getByTestId("submit-button"));
    await waitFor(() => {
      expect(toJSON()).toMatchSnapshot();
    });
  });
  it("should render correct ui for AccountVerificationScreen when verificationError is defined", async () => {
    const route = {
      key: "AccountVerification-wPFrEJG5kjiOrE_Qk2kVs",
      name: "AccountVerification" as "AccountVerification",
      params: { verificationError: "wrong code" },
    };

    const { toJSON, getByTestId } = render(
      <AccountVerificationScreen route={route} />
    );

    fireEvent.press(getByTestId("show-password-icon"));

    fireEvent.changeText(getByTestId("employee-id-input"), "some");
    fireEvent(getByTestId("employee-id-input"), "onSubmitEditing", {
      nativeEvent: { text: "some" },
    });
    fireEvent.changeText(getByTestId("code-input"), "code");
    fireEvent.press(getByTestId("submit-button"));
    await waitFor(() => {
      expect(toJSON()).toMatchSnapshot();
    });
  });
});
