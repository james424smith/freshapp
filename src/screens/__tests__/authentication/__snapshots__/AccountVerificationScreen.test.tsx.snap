// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountVerificationScreen snapshot test should render correct ui for AccountVerificationScreen when verificationError is defined 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "backgroundColor": "#f6f6f6",
      "flex": 1,
      "width": 750,
    }
  }
>
  <RCTScrollView
    collapsable={false}
    contentContainerStyle={
      Object {
        "flex": 1,
      }
    }
    keyboardShouldPersistTaps="handled"
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View>
      <BVLinearGradient
        colors={
          Array [
            4278819773,
            4278213285,
          ]
        }
        endPoint={
          Object {
            "x": 0,
            "y": 1.5,
          }
        }
        locations={
          Array [
            0,
            0.5,
          ]
        }
        startPoint={
          Object {
            "x": 0.1,
            "y": 0,
          }
        }
        style={
          Object {
            "backgroundColor": "#f6f6f6",
            "flex": 1,
            "width": 750,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "justifyContent": "space-between",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-end",
                "paddingTop": 33.5,
              }
            }
          >
            <View
              style={
                Object {
                  "overflow": "hidden",
                }
              }
            >
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "flex-end",
                    "justifyContent": "space-between",
                    "padding": 22.5,
                  }
                }
                testID="exit-button"
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "justifyContent": "center",
                    }
                  }
                >
                  <Text
                    allowFontScaling={false}
                    style={
                      Array [
                        Object {
                          "color": "#242424",
                          "fontSize": 30,
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                        Object {
                          "fontFamily": "Ionicons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                    testID="iconIcon"
                  >
                    ïˆ¥
                  </Text>
                </View>
              </View>
            </View>
          </View>
          <View />
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "marginTop": 13.5,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#2A2E43",
                    "fontFamily": "Roboto-Regular",
                  },
                  Object {
                    "color": "white",
                    "fontSize": 53.5,
                    "fontWeight": "700",
                  },
                ]
              }
            >
              Account verification
            </Text>
          </View>
          <View
            style={
              Object {
                "justifyContent": "center",
                "marginHorizontal": 75,
              }
            }
          >
            <View>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#2A2E43",
                      "fontFamily": "Roboto-Regular",
                    },
                    Object {
                      "color": undefined,
                      "fontSize": 24,
                      "marginBottom": 4,
                      "marginLeft": 30,
                    },
                  ]
                }
              >
                Employee ID
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": undefined,
                  "borderRadius": 30,
                  "height": 86.5,
                  "marginBottom": 22.5,
                  "paddingHorizontal": 10,
                  "width": "100%",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "borderBottomWidth": 0,
                    "borderColor": "#86939e",
                    "flexDirection": "row",
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                    ],
                    "zIndex": 1,
                  }
                }
              >
                <TextInput
                  allowFontScaling={true}
                  autoCapitalize="none"
                  autoCorrect={false}
                  autoFocus={false}
                  blurOnSubmit={false}
                  editable={false}
                  keyboardAppearance="light"
                  keyboardType="numeric"
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  onFocus={[Function]}
                  onSubmitEditing={[Function]}
                  placeholder="Employee ID"
                  placeholderTextColor="#8FAABF"
                  rejectResponderTermination={true}
                  returnKeyType="default"
                  secureTextEntry={false}
                  style={
                    Object {
                      "alignSelf": "center",
                      "color": "#FFF",
                      "flex": 1,
                      "flexDirection": undefined,
                      "fontFamily": "Roboto-Regular",
                      "fontSize": 29.5,
                      "marginLeft": 22.5,
                      "minHeight": 40,
                      "opacity": 0.5,
                      "width": 600,
                    }
                  }
                  testID="employee-id-input"
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
              <Text
                style={
                  Object {
                    "color": "#ff190c",
                    "fontSize": 12,
                    "margin": 5,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": undefined,
                  "borderRadius": 30,
                  "height": 86.5,
                  "marginBottom": 22.5,
                  "paddingHorizontal": 10,
                  "width": "100%",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "borderBottomWidth": 0,
                    "borderColor": "#86939e",
                    "flexDirection": "row",
                    "marginTop": -5,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                    ],
                    "zIndex": 1,
                  }
                }
              >
                <TextInput
                  allowFontScaling={true}
                  autoCapitalize="none"
                  autoCorrect={false}
                  blurOnSubmit={false}
                  editable={true}
                  keyboardAppearance="light"
                  keyboardType="numeric"
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  onFocus={[Function]}
                  placeholder="Email Verification code "
                  placeholderTextColor="#8FAABF"
                  rejectResponderTermination={true}
                  returnKeyType="next"
                  secureTextEntry={false}
                  style={
                    Object {
                      "alignSelf": "center",
                      "color": "#FFF",
                      "flex": 1,
                      "flexDirection": undefined,
                      "fontFamily": "Roboto-Regular",
                      "fontSize": 29.5,
                      "marginLeft": 22.5,
                      "minHeight": 40,
                      "width": 600,
                    }
                  }
                  testID="code-input"
                  underlineColorAndroid="transparent"
                  value="code"
                />
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "height": 40,
                      "justifyContent": "center",
                      "marginVertical": 4,
                      "paddingRight": 4,
                    }
                  }
                >
                  <View
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                    testID="show-password-icon"
                  >
                    <<svg xmlns="http://www.w3.org/2000/svg" width="16" height="10.786" viewBox="0 0 16 10.786"><path d="M10.25,7.383a8.6,8.6,0,0,0-8,5.393,8.63,8.63,0,0,0,16,0A8.6,8.6,0,0,0,10.25,7.383Zm0,8.989a3.6,3.6,0,1,1,3.636-3.6A3.626,3.626,0,0,1,10.25,16.372Zm0-5.754a2.157,2.157,0,1,0,2.182,2.157A2.177,2.177,0,0,0,10.25,10.619Z" transform="translate(-2.25 -7.383)"/></svg>
                      fill="#000"
                      height={20}
                      width={20}
                    />
                  </View>
                </View>
              </View>
              <Text
                style={
                  Object {
                    "color": "#ff190c",
                    "fontSize": 12,
                    "margin": 5,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "paddingLeft": 37.5,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#2A2E43",
                      "fontFamily": "Roboto-Regular",
                    },
                    Object {
                      "color": "#FFFFFF",
                      "fontSize": 20,
                    },
                  ]
                }
              >
                wrong code
              </Text>
            </View>
            <View
              style={
                Object {
                  "paddingTop": 53.5,
                }
              }
            >
              <View
                style={
                  Object {
                    "borderRadius": 25,
                    "overflow": "hidden",
                  }
                }
              >
                <View
                  accessibilityRole="button"
                  accessibilityState={
                    Object {
                      "busy": false,
                      "disabled": true,
                    }
                  }
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                  testID="submit-button"
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": undefined,
                        "borderColor": "#2089dc",
                        "borderRadius": 25,
                        "borderWidth": 0,
                        "flexDirection": "row",
                        "height": 100,
                        "justifyContent": "center",
                        "marginBottom": 26.5,
                        "padding": 8,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": undefined,
                          "fontSize": 30.5,
                          "paddingBottom": 1,
                          "paddingTop": 2,
                          "textAlign": "center",
                        }
                      }
                    >
                      Next
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "paddingBottom": 40,
              }
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                      "justifyContent": "center",
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#2A2E43",
                          "fontFamily": "Roboto-Regular",
                        },
                        Object {
                          "color": "#78849E",
                          "fontSize": 22.5,
                        },
                      ]
                    }
                  >
                    By registering you accept our 
                  </Text>
                  <View
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                    testID="privacy-policy-button"
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "#2A2E43",
                            "fontFamily": "Roboto-Regular",
                          },
                          Object {
                            "color": "#78849E",
                            "fontSize": 22.5,
                            "textDecorationLine": "underline",
                          },
                        ]
                      }
                    >
                      Privacy Policy
                    </Text>
                  </View>
                </View>
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                      "justifyContent": "center",
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#2A2E43",
                          "fontFamily": "Roboto-Regular",
                        },
                        Object {
                          "color": "#78849E",
                          "fontSize": 22.5,
                        },
                      ]
                    }
                  >
                     and 
                  </Text>
                  <View
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                    testID="terms-and-condition-button"
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "#2A2E43",
                            "fontFamily": "Roboto-Regular",
                          },
                          Object {
                            "color": "#78849E",
                            "fontSize": 22.5,
                            "textDecorationLine": "underline",
                          },
                        ]
                      }
                    >
                      Terms & Conditions
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </BVLinearGradient>
    </View>
  </RCTScrollView>
</View>
`;

exports[`AccountVerificationScreen snapshot test should render correct ui for AccountVerificationScreen when verificationError is undefined 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "backgroundColor": "#f6f6f6",
      "flex": 1,
      "width": 750,
    }
  }
>
  <RCTScrollView
    collapsable={false}
    contentContainerStyle={
      Object {
        "flex": 1,
      }
    }
    keyboardShouldPersistTaps="handled"
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View>
      <BVLinearGradient
        colors={
          Array [
            4278819773,
            4278213285,
          ]
        }
        endPoint={
          Object {
            "x": 0,
            "y": 1.5,
          }
        }
        locations={
          Array [
            0,
            0.5,
          ]
        }
        startPoint={
          Object {
            "x": 0.1,
            "y": 0,
          }
        }
        style={
          Object {
            "backgroundColor": "#f6f6f6",
            "flex": 1,
            "width": 750,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "justifyContent": "space-between",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-end",
                "paddingTop": 33.5,
              }
            }
          >
            <View
              style={
                Object {
                  "overflow": "hidden",
                }
              }
            >
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "flex-end",
                    "justifyContent": "space-between",
                    "padding": 22.5,
                  }
                }
                testID="exit-button"
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "justifyContent": "center",
                    }
                  }
                >
                  <Text
                    allowFontScaling={false}
                    style={
                      Array [
                        Object {
                          "color": "#242424",
                          "fontSize": 30,
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                        Object {
                          "fontFamily": "Ionicons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                    testID="iconIcon"
                  >
                    ïˆ¥
                  </Text>
                </View>
              </View>
            </View>
          </View>
          <View />
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "marginTop": 13.5,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#2A2E43",
                    "fontFamily": "Roboto-Regular",
                  },
                  Object {
                    "color": "white",
                    "fontSize": 53.5,
                    "fontWeight": "700",
                  },
                ]
              }
            >
              Account verification
            </Text>
          </View>
          <View
            style={
              Object {
                "justifyContent": "center",
                "marginHorizontal": 75,
              }
            }
          >
            <View>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#2A2E43",
                      "fontFamily": "Roboto-Regular",
                    },
                    Object {
                      "color": undefined,
                      "fontSize": 24,
                      "marginBottom": 4,
                      "marginLeft": 30,
                    },
                  ]
                }
              >
                Employee ID
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": undefined,
                  "borderRadius": 30,
                  "height": 86.5,
                  "marginBottom": 22.5,
                  "paddingHorizontal": 10,
                  "width": "100%",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "borderBottomWidth": 0,
                    "borderColor": "#86939e",
                    "flexDirection": "row",
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                    ],
                    "zIndex": 1,
                  }
                }
              >
                <TextInput
                  allowFontScaling={true}
                  autoCapitalize="none"
                  autoCorrect={false}
                  autoFocus={false}
                  blurOnSubmit={false}
                  editable={false}
                  keyboardAppearance="light"
                  keyboardType="numeric"
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  onFocus={[Function]}
                  onSubmitEditing={[Function]}
                  placeholder="Employee ID"
                  placeholderTextColor="#8FAABF"
                  rejectResponderTermination={true}
                  returnKeyType="default"
                  secureTextEntry={false}
                  style={
                    Object {
                      "alignSelf": "center",
                      "color": "#FFF",
                      "flex": 1,
                      "flexDirection": undefined,
                      "fontFamily": "Roboto-Regular",
                      "fontSize": 29.5,
                      "marginLeft": 22.5,
                      "minHeight": 40,
                      "opacity": 0.5,
                      "width": 600,
                    }
                  }
                  testID="employee-id-input"
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
              <Text
                style={
                  Object {
                    "color": "#ff190c",
                    "fontSize": 12,
                    "margin": 5,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": undefined,
                  "borderRadius": 30,
                  "height": 86.5,
                  "marginBottom": 22.5,
                  "paddingHorizontal": 10,
                  "width": "100%",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "borderBottomWidth": 0,
                    "borderColor": "#86939e",
                    "flexDirection": "row",
                    "marginTop": -5,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                    ],
                    "zIndex": 1,
                  }
                }
              >
                <TextInput
                  allowFontScaling={true}
                  autoCapitalize="none"
                  autoCorrect={false}
                  blurOnSubmit={false}
                  editable={true}
                  keyboardAppearance="light"
                  keyboardType="numeric"
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  onFocus={[Function]}
                  placeholder="Email Verification code "
                  placeholderTextColor="#8FAABF"
                  rejectResponderTermination={true}
                  returnKeyType="next"
                  secureTextEntry={false}
                  style={
                    Object {
                      "alignSelf": "center",
                      "color": "#FFF",
                      "flex": 1,
                      "flexDirection": undefined,
                      "fontFamily": "Roboto-Regular",
                      "fontSize": 29.5,
                      "marginLeft": 22.5,
                      "minHeight": 40,
                      "width": 600,
                    }
                  }
                  testID="code-input"
                  underlineColorAndroid="transparent"
                  value="code"
                />
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "height": 40,
                      "justifyContent": "center",
                      "marginVertical": 4,
                      "paddingRight": 4,
                    }
                  }
                >
                  <View
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                    testID="show-password-icon"
                  >
                    <<svg xmlns="http://www.w3.org/2000/svg" width="16" height="10.786" viewBox="0 0 16 10.786"><path d="M10.25,7.383a8.6,8.6,0,0,0-8,5.393,8.63,8.63,0,0,0,16,0A8.6,8.6,0,0,0,10.25,7.383Zm0,8.989a3.6,3.6,0,1,1,3.636-3.6A3.626,3.626,0,0,1,10.25,16.372Zm0-5.754a2.157,2.157,0,1,0,2.182,2.157A2.177,2.177,0,0,0,10.25,10.619Z" transform="translate(-2.25 -7.383)"/></svg>
                      fill="#000"
                      height={20}
                      width={20}
                    />
                  </View>
                </View>
              </View>
              <Text
                style={
                  Object {
                    "color": "#ff190c",
                    "fontSize": 12,
                    "margin": 5,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "paddingTop": 53.5,
                }
              }
            >
              <View
                style={
                  Object {
                    "borderRadius": 25,
                    "overflow": "hidden",
                  }
                }
              >
                <View
                  accessibilityRole="button"
                  accessibilityState={
                    Object {
                      "busy": false,
                      "disabled": true,
                    }
                  }
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                  testID="submit-button"
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": undefined,
                        "borderColor": "#2089dc",
                        "borderRadius": 25,
                        "borderWidth": 0,
                        "flexDirection": "row",
                        "height": 100,
                        "justifyContent": "center",
                        "marginBottom": 26.5,
                        "padding": 8,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": undefined,
                          "fontSize": 30.5,
                          "paddingBottom": 1,
                          "paddingTop": 2,
                          "textAlign": "center",
                        }
                      }
                    >
                      Next
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "paddingBottom": 40,
              }
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                      "justifyContent": "center",
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#2A2E43",
                          "fontFamily": "Roboto-Regular",
                        },
                        Object {
                          "color": "#78849E",
                          "fontSize": 22.5,
                        },
                      ]
                    }
                  >
                    By registering you accept our 
                  </Text>
                  <View
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                    testID="privacy-policy-button"
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "#2A2E43",
                            "fontFamily": "Roboto-Regular",
                          },
                          Object {
                            "color": "#78849E",
                            "fontSize": 22.5,
                            "textDecorationLine": "underline",
                          },
                        ]
                      }
                    >
                      Privacy Policy
                    </Text>
                  </View>
                </View>
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                      "justifyContent": "center",
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#2A2E43",
                          "fontFamily": "Roboto-Regular",
                        },
                        Object {
                          "color": "#78849E",
                          "fontSize": 22.5,
                        },
                      ]
                    }
                  >
                     and 
                  </Text>
                  <View
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                    testID="terms-and-condition-button"
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "#2A2E43",
                            "fontFamily": "Roboto-Regular",
                          },
                          Object {
                            "color": "#78849E",
                            "fontSize": 22.5,
                            "textDecorationLine": "underline",
                          },
                        ]
                      }
                    >
                      Terms & Conditions
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </BVLinearGradient>
    </View>
  </RCTScrollView>
</View>
`;
